iphone/Maps/Bookmarks/Categories/BMCViewModel/BMCDefaultViewModel.swift

```diff
@@ -129,7 +129,8 @@ extension BMCDefaultViewModel {
     }
 
     categories.append(manager.category(withId: manager.createCategory(withName: name)))
-    view?.insert(at: [IndexPath(row: categories.count - 1, section: section)])
+    categories = getCategories()
```

Comment:
This line is redundant because you append the new item to the `categories` and then reassing the whole property.
You can insert the new category as a first element both in the `category` and the `tableView`:
```swift
    categories.insert(manager.category(withId: manager.createCategory(withName: name)), at: 0)
    view?.insert(at: [IndexPath(row: 0, section: section)])
```

Reply:
Yep, better to insert at the desired postion. Saves unnecessary sorting.